extra/typeorm-model-shim.js

```diff
@@ -20,248 +20,375 @@
 
 /* export */ function Index() {
-    return function () {};
+    return function () { };
 }
 exports.Index = Index;
 
 /* export */ function Unique() {
-    return function () {};
+    return function () { };
 }
 exports.Unique = Unique;
 
 /* export */ function Check() {
-    return function () {};
+    return function () { };
 }
 exports.Check = Check;
 
 /* export */ function Exclusion() {
-    return function () {};
+    return function () { };
 }
 exports.Exclusion = Exclusion;
 
 /* export */ function Generated() {
-    return function () {};
+    return function () { };
 }
 exports.Generated = Generated;
 
 /* export */ function EntityRepository() {
-    return function () {};
+    return function () { };
 }
 exports.EntityRepository = EntityRepository;
+
+
+/* export */ function BaseEntity() { }
+
+BaseEntity.prototype.hasId = function () {
+    return false;
+};
+BaseEntity.prototype.save = function (options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.prototype.remove = function (options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.prototype.softRemove = function (options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.prototype.recover = function (options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.prototype.reload = function () {
+    return new Promise((resolve, reject) => { });
+};
+
+
+BaseEntity.useConnection = function (connection) { };
+BaseEntity.getRepository = function (objectType) {
+    return function () { };
+};
+BaseEntity.target = function () { };
+
+BaseEntity.hasId = function (entity) {
+    return false;
+};
+BaseEntity.getId = function (objectType, entity) {
+    return undefined;
+};
+BaseEntity.createQueryBuilder = function (objectType, alias) {
+    return function () { };
+};
+BaseEntity.create = function (objectType, entityLike = null) {
+    if (entityLike && Array.isArray(entityLike)) {
+        return [new objectType()];
+    } else {
+        return new objectType();
+    }
+};
+BaseEntity.merge = function (objectType, mergeIntoEntity, ...entityLikes) {
+    return mergeIntoEntity;
+};
+BaseEntity.preload = function (objectType, entityLike) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.save = function (objectType, entities, options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.save = function (objectType, entity, options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.remove = function (objectType, entities, options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.remove = function (objectType, entity, options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.softRemove = function (objectType, entities, options) {
+    return new Promise((resolve, reject) => { });
+};
+BaseEntity.softRemove = function (objectType, entity, options) {
+    return new Promise((resolve, reject) => { });
```

Comment:
This promise never fullfills. I think you should either use `Promise.reject(new Error)` or `Promise.resolve()`. Never fulfilling promises are extremely hard to debug.